src/core/auth/qgsauthcertutils.cpp

```diff
@@ -1017,3 +1017,111 @@ QList<QPair<QSslError::SslError, QString> > QgsAuthCertUtils::sslErrorEnumString
                          QgsAuthCertUtils::sslErrorEnumString( QSslError::CertificateBlacklisted ) );
   return errenums;
 }
+
+QList<QSslError> QgsAuthCertUtils::validateCertChain( const QList<QSslCertificate> &certificateChain,
+    const QString &hostName,
+    bool trustRootCa )
+{
+  QList<QSslError> sslErrors;
+  QList<QSslCertificate> trustedChain;
+  // Filter out all CAs that are not trusted from QgsAuthManager
+  for ( const auto &cert : certificateChain )
+  {
+    bool untrusted = false;
+    for ( const auto &untrustedCert : QgsAuthManager::instance()->getUntrustedCaCerts() )
```

Comment:
I know we spoke about calling QgsAuthManager::instance() in this utils class of static functions, but what's your thoughts on adding an `untrustedCas` parameter? Doing so will also allow us to define exactly _what certs_ are untrusted, instead of always assuming it is `QgsAuthManager::instance()->getUntrustedCaCerts`.

Basically, I think we should take this opportunity to remove those calls to QgsAuthManager::instance() in this utils class.

Reply:
Nevermind. Too difficult to remove all usages. Leaving as is.